Backport patches for rdma cm patches

Backport patch for ipv6_dev_get_saddr for kernel v2.6.26.

Signed-of-by: David Wilder <dwilder@us.ibm.com>
---
diff -Naurp ofa_kernel.orig/drivers/infiniband/core/addr.c ofa_kernel/drivers/infiniband/core/addr.c
--- ofa_kernel.orig/drivers/infiniband/core/addr.c	2009-12-09 12:42:24.000000000 -0600
+++ ofa_kernel/drivers/infiniband/core/addr.c	2009-12-09 12:45:47.000000000 -0600
@@ -250,7 +250,7 @@ static int addr6_resolve(struct sockaddr
 		goto put;
 
 	if (ipv6_addr_any(&fl.fl6_src)) {
-		ret = ipv6_dev_get_saddr(&init_net, ip6_dst_idev(dst)->dev,
+		ret = ipv6_dev_get_saddr(ip6_dst_idev(dst)->dev,
 					 &fl.fl6_dst, 0, &fl.fl6_src);
 		if (ret)
 			goto put;


